sunpycube/cube/tests/test_NDCube.py

```diff
@@ -0,0 +1,208 @@
+    (cubem[:, :, :].dimensions, DimensionPair(
+        dimensions=u.Quantity((2, 3, 4), unit=u.pix), axes=['HPLN-TAN', 'HPLT-TAN', 'WAVE'])),
+    (cubem[:, 1, 1].dimensions, DimensionPair(
+        dimensions=u.Quantity((2,), unit=u.pix), axes=['HPLN-TAN'])),
+    (cubem[:, 1, 0:2].dimensions, DimensionPair(
+        dimensions=u.Quantity((2, 2), unit=u.pix), axes=['HPLN-TAN', 'WAVE'])),
+    (cubem[:, 1, :].dimensions, DimensionPair(dimensions=u.Quantity(
+        (2, 4), unit=u.pix), axes=['HPLN-TAN', 'WAVE'])),
+    (cubem[1, :, 1].dimensions, DimensionPair(
+        dimensions=u.Quantity((3,), unit=u.pix), axes=['HPLT-TAN'])),
+    (cubem[1, :, 0:2].dimensions, DimensionPair(
+        dimensions=u.Quantity((3, 2), unit=u.pix), axes=['HPLT-TAN', 'WAVE'])),
+    (cubem[1, :, :].dimensions, DimensionPair(dimensions=u.Quantity(
+        (3, 4), unit=u.pix), axes=['HPLT-TAN', 'WAVE'])),
+    (cubem[1, 1, 1].dimensions, DimensionPair(dimensions=u.Quantity((), unit=u.pix), axes=[])),
+    (cubem[1, 1, 0:2].dimensions, DimensionPair(
+        dimensions=u.Quantity((2,), unit=u.pix), axes=['WAVE'])),
+    (cubem[1, 1, :].dimensions, DimensionPair(
+        dimensions=u.Quantity((4,), unit=u.pix), axes=['WAVE'])),
+    (cube[:, :, 1].dimensions, DimensionPair(dimensions=u.Quantity(
+        (2, 3), unit=u.pix), axes=['HPLT-TAN', 'WAVE'])),
+    (cube[:, :, 0:2].dimensions, DimensionPair(dimensions=u.Quantity(
+        (2, 3, 2), unit=u.pix), axes=['HPLT-TAN', 'WAVE', 'TIME'])),
+    (cube[:, :, :].dimensions, DimensionPair(dimensions=u.Quantity(
+        (2, 3, 4), unit=u.pix), axes=['HPLT-TAN', 'WAVE', 'TIME'])),
+    (cube[:, 1, 1].dimensions, DimensionPair(
+        dimensions=u.Quantity((2,), unit=u.pix), axes=['HPLT-TAN'])),
+    (cube[:, 1, 0:2].dimensions, DimensionPair(
+        dimensions=u.Quantity((2, 2), unit=u.pix), axes=['HPLT-TAN', 'TIME'])),
+    (cube[:, 1, :].dimensions, DimensionPair(dimensions=u.Quantity(
+        (2, 4), unit=u.pix), axes=['HPLT-TAN', 'TIME'])),
+    (cube[1, :, 1].dimensions, DimensionPair(dimensions=u.Quantity((3,), unit=u.pix), axes=['WAVE'])),
+    (cube[1, :, 0:2].dimensions, DimensionPair(
+        dimensions=u.Quantity((3, 2), unit=u.pix), axes=['WAVE', 'TIME'])),
+    (cube[1, :, :].dimensions, DimensionPair(
+        dimensions=u.Quantity((3, 4), unit=u.pix), axes=['WAVE', 'TIME'])),
+    (cube[1, 1, 1].dimensions, DimensionPair(dimensions=u.Quantity((), unit=u.pix), axes=[])),
+    (cube[1, 1, 0:2].dimensions, DimensionPair(
+        dimensions=u.Quantity((2,), unit=u.pix), axes=['TIME'])),
+    (cube[1, 1, :].dimensions, DimensionPair(dimensions=u.Quantity((4,), unit=u.pix
+                                                                   ), axes=['TIME'])),
+])
+def test_slicing_third_axis(test_input, expected):
+    assert test_input[1] == expected[1]
+    assert np.all(test_input[0].value == expected[0].value)
+    assert test_input[0].unit == expected[0].unit
```

Comment:
The uncertainty is an optional attribute which is an array of the same shape as the data attribute giving the uncertainty of each pixel/time, etc.  So you can think of it as though it's another data array.

Good work on the mask and wcs checks :)